system-tests/ft-suite/src/test/kotlin/com/hotels/styx/resiliency/OriginResourcesSpec.kt

```diff
@@ -0,0 +1,161 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.resiliency
+
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.hotels.styx.api.HttpHeaderNames.HOST
+import com.hotels.styx.api.HttpRequest.get
+import com.hotels.styx.api.HttpResponseStatus.OK
+import com.hotels.styx.client.StyxHttpClient
+import com.hotels.styx.server.HttpConnectorConfig
+import com.hotels.styx.servers.MockOriginServer
+import com.hotels.styx.support.StyxServerProvider
+import com.hotels.styx.support.proxyHttpHostHeader
+import com.hotels.styx.support.wait
+import io.kotlintest.Spec
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.StringSpec
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.nio.charset.StandardCharsets.UTF_8
+import java.nio.file.Files.copy
+import java.nio.file.Path
+import java.nio.file.StandardCopyOption.REPLACE_EXISTING
+
+class OriginResourcesSpec : StringSpec() {
+    val count = 20
+
+    val mockServer = MockOriginServer.create("", "", 0, HttpConnectorConfig(0))
+            .start()
+            .stub(WireMock.get(WireMock.urlMatching("/.*")), WireMock.aResponse()
+                    .withStatus(200)
+                    .withBody("mock-server-01"))
+
+
+    init {
+        "Cleans up threads after use" {
+            "---" + (1..count)
+                    .map { appDeclaration("aaa-$it") }
+                    .joinToString("\n")
+                    .let { writeConfig(styxOriginsFile, it) }
+
+            // Wait for styx to pick up the configuration
+            Thread.sleep(2000)
```

Comment:
We should use `io.kotlintest.eventually` instead of a long sleep.

Reply:
See above.